apps/src/templates/progress/ProgressLessonTeacherInfo.jsx

```diff
@@ -34,46 +37,74 @@ const styles = {
 class ProgressLessonTeacherInfo extends React.Component {
   static propTypes = {
     lesson: lessonType.isRequired,
+    levelUrl: PropTypes.string.isRequired,
 
     // redux provided
-    sectionId: PropTypes.string,
+    section: sectionShape,
     scriptAllowsHiddenStages: PropTypes.bool.isRequired,
     hiddenStageState: PropTypes.object.isRequired,
     scriptName: PropTypes.string.isRequired,
     hasNoSections: PropTypes.bool.isRequired,
-    toggleHiddenStage: PropTypes.func.isRequired
+    toggleHiddenStage: PropTypes.func.isRequired,
+    showGoogleClassroomButton: PropTypes.bool.isRequired
   };
 
   onClickHiddenToggle = value => {
-    const {scriptName, sectionId, lesson, toggleHiddenStage} = this.props;
+    const {scriptName, section, lesson, toggleHiddenStage} = this.props;
+    const sectionId = section.id.toString();
     toggleHiddenStage(scriptName, sectionId, lesson.id, value === 'hidden');
-    firehoseClient.putRecord({
-      study: 'hidden-lessons',
-      study_group: 'v0',
-      event: value,
-      data_json: JSON.stringify({
-        script_name: scriptName,
-        section_id: sectionId,
-        lesson_id: lesson.id,
-        lesson_name: lesson.name
-      })
-    });
+    firehoseClient.putRecord(
+      {
+        study: 'hidden-lessons',
+        study_group: 'v0',
+        event: value,
+        data_json: JSON.stringify(this.firehoseData())
+      },
+      {includeUserId: true}
+    );
+  };
+
+  firehoseData = () => {
```

Comment:
Nit (trying to remember) we decided to bind these in the constructor, right?

Reply:
hah! we most certainly did! i'll also update `onClickHiddenToggle` while i'm at it. ;-)